File: examples/js/raml10-resolution.js
/**
 * Example of resolving/expanding RAML 1.0 data types.
 */
const wap = require('webapi-parser').WebApiParser
const path = require('path')

const ramlStr = `#%RAML 1.0 Library
types:
  User:
    type: object
    properties:
      firstName: string
      lastName:  string
  Admin:
    type: User
    properties:
      permissions:
        type: array
        items: string
`

async function main () {
  // Parse RAML 1.0 string
  const model = await wap.raml10.parse(ramlStr)

  // Resolve/expand types defined in the document
  const resolved = await wap.raml10.resolve(model)

  // Generate output file with the resolved model RAML content
  const outPath = path.join(__dirname, './generated.raml')
  await wap.raml10.generateFile(resolved, `file://${outPath}`)
  // Or
  // const outputRaml = await wap.raml10.generateString(resolved)
}

main()

File: examples/java/src/main/java/co/acme/model/Raml10Resolution.java
package co.acme.model;

import webapi.Raml10;
import webapi.WebApiDocument;

import java.util.concurrent.ExecutionException;

public class Raml10Resolution {

  // Example of resolving RAML 1.0 file
  public static void resolveApi() throws InterruptedException, ExecutionException {
    // Parse the file
    WebApiDocument doc = (WebApiDocument) Raml10.parse("file://../api-specs/raml/api-with-types.raml").get();

    // Resolve parsed model
    WebApiDocument resolved = (WebApiDocument) Raml10.resolve(doc).get();

    // Generate RAML 1.0 string from the resolved model and log it
    String output = Raml10.generateString(resolved).get();
    System.out.println("Generated Raml10 string:\n" + output);
  }
}
